lib/Sema/CodeSynthesis.cpp

```diff
@@ -2405,9 +2405,16 @@ static void configureDesignatedInitAttributes(TypeChecker &tc,
 
   // If the superclass has its own availability, make sure the synthesized
   // constructor is only as available as its superclass's constructor.
-  if (superclassCtor->getAttrs().hasAttribute<AvailableAttr>()) {
+  {
```

Comment:
The condition should still be here. (This version isn't wrong, but it is less efficient.)